backend/src/apiserver/template/tekton_template.go

```diff
@@ -331,8 +331,8 @@ func (t *Tekton) injectArchivalStep(workflow util.Workflow, artifactItemsJSON ma
 					t.getObjectFieldSelector("PIPELINERUN", "metadata.labels['tekton.dev/pipelineRun']"),
 					t.getObjectFieldSelector("PODNAME", "metadata.name"),
 					t.getObjectFieldSelector("NAMESPACE", "metadata.namespace"),
-					t.getSecretKeySelector("AWS_ACCESS_KEY_ID", "mlpipeline-minio-artifact", "accesskey"),
```

Comment:
For the S3 access and secret key, we want to load it from the secret instead of putting it as raw EnvVar because we don't want to expose the credential values for anyone who can read the pod spec.

Reply:
That's the point of this change. I'm delegating to the kustomize manifests to inject the env var (which can be from a secret) instead of having a hard-coded secret name.

If you believe this is not a good solution, do you have any suggestions?

Reply:
The reason we need it to load the environment variable from secret because the Kubeflow multi-user integration will populate the object storage credentials as secret into each namespace. Right now I think the end to end test got stuck because the new pods are not getting the correct AWS_ACCESS_KEY_ID information as raw value environment variable in k8s.

Reply:
How does multi-user integration works? Does kfp-tekton create a Deployment object per namespace?

Reply:
kfp-tekton is not the one creating the user config. For full Kubeflow deployment, there will be a profile controller that initialize user configuration for each new user namespace such as the minio secret information. Is it possible to keep using the `getSecretKeySelector` function but parameterize the secret name `mlpipeline-minio-artifact` and key name `accesskey` instead?

Reply:
The user minio secret can be configured in the profile controller client over here.
https://github.com/kubeflow/kfp-tekton/blob/master/manifests/kustomize/base/installs/multi-user/pipelines-profile-controller/sync.py#L366-L379

Reply:
> kfp-tekton is not the one creating the user config. For full Kubeflow deployment, there will be a profile controller that initialize user configuration for each new user namespace such as the minio secret information. Is it possible to keep using the `getSecretKeySelector` function but parameterize the secret name `mlpipeline-minio-artifact` and key name `accesskey` instead?

That was my initial plan, but I did that to be parametrized from the kustomize manifests. I'll setup a multi-user environment to understand how it works, and will use my custom image to see how it behaves.

Reply:
sure that sounds good to me

Reply:
Cool, let me know if we can merge this PR and I'll resolve this conversation.

Reply:
yes, we just need to set back the default values such as `mlpipeline-minio-artifact` for these secret name. So it won't break our current end to end test

Reply:
So I sent the update, let me know if that works.